Model/Api/OrderManagement.php

```diff
@@ -189,6 +189,33 @@ public function manage(
         }
     }
 
+    /**
+     * Handle custom checkboxes if request has them
+     *
+     * @param string $displayId
+     */
+    private function handleCheckboxes($displayId) {
+        $request = $this->request->getBodyParams();
+        if (!isset($request['checkboxes']) || !$request['checkboxes']) {
+            return;
+        }
+        $comment = '';
+        $needSubscribe = false;
+        foreach ($request['checkboxes'] as $checkbox) {
+            if ($checkbox['category'] == 'NEWSLETTER') {
+                $needSubscribe = true;
+            }
+            $comment .= '<br>'.$checkbox['text'].': '.($checkbox['value'] ? 'Yes' : 'No');
+        }
+        if ($comment) {
+            $commentPrefix = 'BOLTPAY INFO :: checkboxes';
+            $this->orderHelper->addCommentToStatusHistoryIfNotExists($displayId, $commentPrefix.$comment, $commentPrefix);
```

Comment:
I was looking at https://devdocs.magento.com/redoc/2.3/admin-rest-api.html#operation/salesOrderRepositoryV1SavePut. It looks like there is a field called `customer_note` what is the difference between that and order comments?

Reply:
**customer_note** is a field for information that customers directly provide for us like "leave this on my porch". Opposite **order comments** for any information we want to save with order.
By default, only merchant staff see the order comment, but we can change it for the specific comment. **customer_note** is available for both staff and customers.
Am I correct @danac-gs

Reply:
Actually, it seems like, although a bit confusing, that customer_note is currently only used when creating back office orders and the entered data is not visible to the customer, only to the merchant in the store backend. The preferred way of adding order comments is via addStatusHistory to sales_order_status_history table, the way we do it.

Reply:
i  agree with @danac-gs